### ISSUE TITLE
Segmentation fault when scrolling past link with line break

### ISSUE THREAD

**Issue by OliverWarrington** [nvim-neorg/neorg#369]
### Prerequisites

- [X] I have read [contribution guidelines](https://github.com/vhyrro/neorg/blob/main/docs/CONTRIBUTING.md)
- [X] I am using the latest version of the plugin
- [X] I am using either 0.6 stable or the latest compiled neovim version

### Neovim Version

NVIM v0.7.0-dev+1046-gaff0ddd78 Build type: Release

### Neorg setup

require('neorg').setup {
    -- Tell Neorg what modules to load
    load = {
        ["core.defaults"] = {}, -- Load all the default modules
        ["core.norg.concealer"] = {
            config = {
                markup_preset = "safe"
            }
        }, -- Allows for use of icons
        ["core.norg.journal"] = {}, -- Allows for use of icons
        ["core.presenter"] = {}, -- Allows for use of icons
        ["core.norg.qol.toc"] = {},
        ["core.keybinds"] = { -- Configure core.keybinds
            config = {
                default_keybinds = true, -- Generate the default keybinds
                neorg_leader = "<Leader>o" -- This is the default if unspecified
            }
        },
        ["core.integrations.telescope"] = {},
        ["core.norg.completion"] = {
            config = {
                engine = "nvim-cmp"
            }
        },
        ["core.norg.dirman"] = { -- Manage your directories with Neorg
            config = {
                workspaces = {
                    my_workspace = "~/work"
                }
            }
        }
    },
}

### Actual behavior

Scrolling past a link that has been split over a line causes a segmentation fault. This originally happened with wrapped text inside a norg file.

I disabled the concealer for the video but this happens with the concealer on both safe and brave presets. 

[](https://user-images.githubusercontent.com/48333946/153657496-b5fe3f07-732c-43fc-adeb-a50e33caa25c.mov)

### Expected behavior

No fault!

### Steps to reproduce

Insert a link, split with a line break, scroll one line past the link.

### Help

Yes, but I don't know how to start. I would need guidance (check question below)

### Implementation help

I would have no idea where to start, always happy to help if I can though!

**Comment by vhyrro** [comment]
Oh my, great catch!

We've been looking into this for a few hours in total now but it seems that the problem is more serious than we anticipated :flushed: 

Here's the most minimal reproducable crash example i've managed to create:
```
{:x:* 
}
```

Initially we just thought this was a problem with our treesitter parser crashing, and soo we fixed the bug in our TS parser that caused errors when putting a newline in a link. What we did not expect though is that the crash still occurs, and - most importantly - only crashes when putting your cursor over the link :thinking: 

We believe this may actually be a very weird one-off case in the treesitter implementation within Neovim, but we're gonna need a little more time to figure this one out fully. Disabling treesitter highlights (through `:TSBufDisable highlight`) actually stops the segfaults.

We are examining the problem though! If we do find a fix I'll be sure to let you know :)
